documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+
+Under `Users`, click on the `View all users` button and you will see two users defined - `alice` and `bob`. `alice` is a member of the `ClusterManager Group`, and `bob` is a member of `ClusterManager-my-cluster Group`.
+In Keycloak the users can be stored in an LDAP identity provider.
+
+Under `Roles` we can see several realm roles which can be used to mark users or clients as having some permissions.
+Roles are a concept analogous to groups. They are usually used to 'tag' users as playing organisational roles and having permissions that pertain to it.
+In Keycloak the roles cannot be stored in an LDAP identity provider - if that is your requirement then you should use groups instead.
+
+Under `Clients` we can see some additional clients configured - `kafka`, `kafka-cli`, `team-a-client`, `team-b-client`.
+The client with client id `kafka` is used by Kafka Brokers to perform the necessary OAuth2 communication for access token validation,
+and to authenticate to other Kafka Broker instances using OAuth2 client authentication.
+This client also contains Authorization Services resource definitions, policies and authorization scopes used to perform authorization on the Kafka Brokers.
+
+The client with client id `kafka-cli` is a public client that can be used by the Kafka command line tools when authenticating with username and password to obtain an access token or a refresh token.
+
+Clients `team-a-client`, and `team-b-client` are confidential clients representing services with partial access to certain Kafka topics.
+
+The authorization configuration is defined in the `kafka` client under `Authorization` tab.
+This tab becomes visible when `Authorization Enabled` is turned on under the `Settings` tab.
+
+
+## Authorization Services - Resources, Authorization Scopes, Policies and Permissions
+
+`Keycloak Authorization Services` use several concepts that together take part in defining, and applying access control to resources.
+
+`Resources` define _what_ we are protecting from unauthorized access.
+Each resource can contain a list of `authorization scopes` - actions that are available on the resource, so that permission on a resource can be granted for one or more actions only.
+`Policies` define the groups of users we want to target with permissions. Users can be targeted based on group membership, assigned roles, or individually.
+Finally, the `permissions` tie together specific `resources`, `action scopes` and `policies` to define that 'specific users U can perform certain actions A on the resource R'.
+
+You can read more about `Keycloak Authorization Services` on link:https://www.keycloak.org/docs/latest/authorization_services/index.html[project's web site], but note that the documentation there is focused primarily on using the Authorization Services to protect web resources, whereas here we're using it to protect access to Kafka.
```

Comment:
```suggestion
For more information about Keycloak Authorization Services, see {keycloak-authorization-services}, but note that the Authorization Services documentation is focused primarily on using the Authorization Services to protect web resources rather than access to Kafka.
```

Reply:
We have an attribute for this link.